internal/temporal/worker_deployment.go

```diff
@@ -0,0 +1,208 @@
+// Unless explicitly stated otherwise all files in this repository are licensed under the MIT License.
+//
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2024 Datadog, Inc.
+
+package temporal
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"strings"
+	"time"
+
+	"go.temporal.io/api/enums/v1"
+	"go.temporal.io/api/serviceerror"
+	temporalClient "go.temporal.io/sdk/client"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	temporaliov1alpha1 "github.com/DataDog/temporal-worker-controller/api/v1alpha1"
+)
+
+// VersionInfo contains information about a specific version
+type VersionInfo struct {
+	VersionID      string
+	Status         temporaliov1alpha1.VersionStatus
+	DrainedSince   *time.Time
+	RampPercentage float32
+	TaskQueues     []temporaliov1alpha1.TaskQueue
+	TestWorkflows  []temporaliov1alpha1.WorkflowExecution
+}
+
+// TemporalWorkerState represents the state of a worker deployment in Temporal
+type TemporalWorkerState struct {
+	DefaultVersionID     string
+	VersionConflictToken []byte
+	RampingVersionID     string
+	RampPercentage       float32
+	RampingSince         *metav1.Time
```

Comment:
Does `RampingSince` indicate the time that the first nonzero ramp value for the current ramping version was enabled, or the time that the ramp value was last modified?

A godoc comment clarifying this would be nice.

Reply:
It's when `RampingVersionID` was last set. Added a comment.